### PR TITLE
Add Expanded view for Time List

### LINKED ISSUE #7377: Allow compact view for time list for Situational Awareness dashboards
**Describe the solution you'd like**
As part of the situational awareness work, we need a compact view of the time list view so it can be included in a dashboard.
This compact view will have minimum needed information and widgets for easier visualization of activities.

The design for compact view can be found on confluence page "Situational Awareness Needs and Design"

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes #7377 

### Describe your changes:
Adds a new component that displays the timelist view in compact mode provided an option `isCompact: true`
Adds inspector configuration to set compact view enabled/disabled

TODO:
- [x] @charlesh88 and @shefalijoshi Need to figure out some details of the item color dot, countdown label, start/end times and progress pie.
- [x] @shefalijoshi Currently hardcoded time Relation and itemState css - this needs to be updated from the plan after merging #7370 
- [x] @charlesh88 CSS final mile.
- [x] @charlesh88 and @shefalijoshi Wire up time hero context strings (STARTS, ENDS, etc.).
- [x] @charlesh88 and @shefalijoshi Finish wiring up time vs. status evaluations, like Incomplete and Overdue. This is needed to control the proper display of icons and the progress pie element.
- [ ] @shefalijoshi Clicking an Activity should select the "Activity" tab in the Inspector.
- [x] Activities should show "STARTS/ENDS/ENDED" depending on their time - inferred states (OVERDUE/RUNNING LONG) should only show if the Activity is not started or in progress, respectively. 1/30/24: "OVERDUE" is not displaying for current Activities that are not started.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [ ] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

**Comment by charlesh88** [comment]
There was a bit of confusion on naming: 'Compact' should refer to the current Time List view:
image

'Expanded' is the name for the new style:
image

**Review by ozyx — CHANGES_REQUESTED** [review]
Reviewed and have some comments regarding the structure of the Vue components and some anti-patterns that we ought not to proliferate due to major performance issues-- we can nip this in the bud straight away with a bit of restructuring.

This also needs tests

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue]: Mutables does take care of this but it happens via `openmct.object.mutate` AFAIK.

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:30]: I believe the intention here is not to apply the change till it is saved. I don't think this was done because the change couldn't take effect immediately.

File: src/plugins/timelist/CompactViewItem.vue
Comment (shefalijoshi) [reply src/plugins/timelist/CompactViewItem.vue]: Yes agreed, this is going to change after the activity states PR is merged.

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (ozyx) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:30]: Why though?

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (ozyx) [inline src/plugins/timelist/inspector/TimelistPropertiesView.vue:40]: since there are only 2 options, we should probably just hardcode this instead of using the v-for

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (ozyx) [inline src/plugins/timelist/inspector/TimelistPropertiesView.vue:46]:
```suggestion
            {{ isExpanded ? 'Expanded' : 'Compact' }}
```

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (ozyx) [inline src/plugins/timelist/inspector/TimelistPropertiesView.vue:108]: this seems overly complicated

**Review by ozyx — CHANGES_REQUESTED** [review]

**Comment by charlesh88** [comment]
Pushed some markup and CSS changes.

### Todos
Shefali
- [ ] Start and end times need to be just hh:mm:ss if date is same as current date.
- [ ] Duration should be in format ##h ##m, not hh:mm:ss.
- [ ] Progress pie should only show if Activity is in-progress.
- [ ] Events (0 duration) should display "Event" in the duration, and not display the End time.

Charles
- [ ] Styling for Activity status.

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:108]: How so?

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:46]: I didn't want to use the hardcoded values here. I'll use a computed property instead.

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:40]: I'd prefer not to hard code this. Is there an issue with using the options like this?

File: src/plugins/timelist/CompactViewItem.vue
Comment (shefalijoshi) [reply src/plugins/timelist/CompactViewItem.vue]: I don't understand. This PR depends on #7370. When that PR is merged, this logic is going to change. It was stubbed in so that Charles and I could work simultaneously on this PR.

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:121]: Done.

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:30]: Because it is jarring to the user. @charlesh88 Can comment to this. If he's OK changing this behavior I have no issues making it consistent.

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (ozyx) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:40]: In my opinion it's much harder to read the code this way as the computed property, data property, and const array of objects all obfuscate the values. Hard-coding isn't always bad! :)

Another thought: this is a binary option, so from a web design perspective it should really be a checkbox, no?
@charlesh88 thoughts on that?

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (ozyx) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:104]: We're declaring an array of objects, assigning it to a reactive data property (it gets wrapped in a Proxy), and generating a computed value to solve the problem of hardcoding 2 strings in a ternary statement

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (ozyx) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:30]: Up to @charlesh88. [He's the one who added it in](https://github.com/nasa/openmct/commit/a99a6b3c828d1de38087c8218007c8445e8a501e) :)

I'll break this out as a separate issue since it's unrelated to this PR, though.

File: src/plugins/timelist/CompactViewItem.vue
Comment (ozyx) [reply src/plugins/timelist/CompactViewItem.vue]: Ah, gotcha. If I understand correctly you're saying we'll fix this in #7370?

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue:40]: It WAS a checkbox. Hehe. We changed it to a dropdown.

File: src/plugins/timelist/CompactView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/CompactView.vue]: See the change made in this commit: https://github.com/nasa/openmct/pull/7378/commits/15acbc9f5d4902fb96eee8dbd8bbcafea3678cce

File: src/plugins/timelist/inspector/TimelistPropertiesView.vue
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/TimelistPropertiesView.vue]: I've just gone ahead and hard coded this in the interest of time.

File: e2e/tests/functional/planning/timelist.e2e.spec.js
Comment (unlikelyzero) [inline e2e/tests/functional/planning/timelist.e2e.spec.js:143]: whoa

**Review by charlesh88 — CHANGES_REQUESTED** [review]
The merge with master (?) had some no-bueno effects:
- [x] Selection is broken and doesn't apply `s-selected` to the selected item, or sometimes only applies it briefly before it goes away. 
- [x] Clicking an activity in both views will sometimes be ignored, and the user must click again one or more times.
- [ ] The progress pie graphic is now broken and doesn't animate properly: 
Activity broken pie display

Was seen, but now can't duplicate after pushing af161f931745fd36024e261e816133aaec7acc45:
- [x] Selecting items in either compact or expanded view, then navigating to the other view, will sporadically result in the Inspector always displaying the same Activity title but with the correct timestamps and Activity statuses for the clicked Activity. In compact view, clicking an Activity, then changing the table sort order, will cause the same problem. Refreshing will fix this, but navving away and back will not. UPDATE: after a fix for undefined filter values, this can no longer be duplicated.
- [x] Refreshing while on either view will display "Layout Item" in the Inspector, with no tabs: 
image

File: src/plugins/timelist/TimelistComponent.vue
Comment (ozyx) [inline src/plugins/timelist/TimelistComponent.vue:301]: this can just be a local variable since it's never used again. should we also handle the case where the object is not found?

File: src/plugins/timelist/TimelistComponent.vue
Comment (ozyx) [inline src/plugins/timelist/TimelistComponent.vue:369]: noice

File: src/plugins/timelist/TimelistComponent.vue
Comment (ozyx) [inline src/plugins/timelist/TimelistComponent.vue:418]: I think we might just be able to do this:

```js
this.planActivities.push(...this.planData[key]);
```

Unless we need a deep clone? I don't think we do.

File: src/plugins/timelist/inspector/svg-progress.js
Comment (ozyx) [inline src/plugins/timelist/inspector/svg-progress.js:1]: We should make this a Vue component on the next pass 😉

**Review by ozyx — CHANGES_REQUESTED** [review]
Alright this is looking good! I've left a bunch of suggestions, mostly small stuff-- but we really, _really_ ought to break those `item` objects up into individual properties or we're losing out on a lot of free perf benefits.

File: src/plugins/timelist/TimelistComponent.vue
Comment (shefalijoshi) [reply src/plugins/timelist/TimelistComponent.vue:418]: Hmm. I think we do? I'd like to leave it for now if that's ok? I'll come back and test it later.

File: src/plugins/timelist/TimelistComponent.vue
Comment (shefalijoshi) [reply src/plugins/timelist/TimelistComponent.vue:301]: There is a missing object interceptor that will create this object if it's not found.

File: src/plugins/timelist/TimelistComponent.vue
Comment (shefalijoshi) [reply src/plugins/timelist/TimelistComponent.vue:47]: Hmm can I do that since it requires `headerItem.property`?

File: src/plugins/timelist/ExpandedViewItem.vue
Comment (shefalijoshi) [reply src/plugins/timelist/ExpandedViewItem.vue:128]: Actually, I looked at this and this refactor prevents re-rendering entire list. It only re-renders the since countdown update.

File: src/plugins/timelist/ExpandedViewItem.vue
Comment (shefalijoshi) [reply src/plugins/timelist/ExpandedViewItem.vue:254]: Done

File: src/plugins/timelist/ExpandedViewItem.vue
Comment (shefalijoshi) [reply src/plugins/timelist/ExpandedViewItem.vue:142]: Done

File: src/plugins/plan/inspector/components/PlanActivitiesView.vue
Comment (ozyx) [inline src/plugins/plan/inspector/components/PlanActivitiesView.vue:132]: it doesn't seem like any of this was reactive? which part was proxied ?

File: src/plugins/timelist/TimelistComponent.vue
Comment (ozyx) [reply src/plugins/timelist/TimelistComponent.vue:420]: Nah, let's change this. JSON parse / stringify is costly anyway.

File: src/plugins/timelist/TimelistComponent.vue
Comment (ozyx) [reply src/plugins/timelist/TimelistComponent.vue:303]: OK. We should still make activityStatesObject a local variable since it's only used in this block

File: src/plugins/timelist/ExpandedViewItem.vue
Comment (ozyx) [reply src/plugins/timelist/ExpandedViewItem.vue:128]: That is unfortunately not the case. Passing objects around like this will cause excessive re-renders-- we've spent a lot of time identifying this pattern (and I presented on the pitfalls of this pattern as part of the Vue 3 retrospective learning session).

Note the re-renders highlighted in green by Vue 3 devtools:


https://github.com/nasa/openmct/assets/9259993/4149cc22-6ef8-4eee-97c5-4a8ceb4a132a

This should be a very easy fix-- we should take care of this now rather than later. Instead of passing in `item`, we should pass in each of item's properties as individual props so Vue can selectively render.

**Review by ozyx — CHANGES_REQUESTED** [review]

File: src/plugins/plan/inspector/components/PlanActivitiesView.vue
Comment (shefalijoshi) [reply src/plugins/plan/inspector/components/PlanActivitiesView.vue:132]: The the selection. Either way this fixes the problem.

File: src/plugins/timelist/ExpandedViewItem.vue
Comment (shefalijoshi) [reply src/plugins/timelist/ExpandedViewItem.vue]: OK I've destructured the props for the ExpandedViewItem. I did not do it for the ListItem though since that's a reusable component, and once the expanded view is shipped nobody is going to use the compact view.
I can create an issue to create a separate component for the ListItem at a later time.

Comment (ozyx) [reply src/plugins/timelist/ExpandedViewItem.vue]: Awesome! I agree that we should leave `ListItem` alone in this PR. I just wanted to make sure we're not creating more tech debt for us to resolve later

File: src/plugins/timelist/ExpandedViewItem.vue
Comment (ozyx) [inline src/plugins/timelist/ExpandedViewItem.vue:147]: This is great! Do you think name, start, end, duration, and countdown, should be required props?

File: src/plugins/timelist/ExpandedViewItem.vue
Comment (ozyx) [inline src/plugins/timelist/ExpandedViewItem.vue:155]: nice

File: src/plugins/timelist/ExpandedViewItem.vue
Comment (ozyx) [inline src/plugins/timelist/ExpandedViewItem.vue:246]: oh weird. where's this defined?

**Review by ozyx — APPROVED** [review]
Good stuff. Awesome job fixing the re-renders-- initial testing shows the list is much more performant 👏 

This all LGTM. I have a few non-blocking comments/questions. nothing urgent:)

**Review by charlesh88 — CHANGES_REQUESTED** [review]
Much better!  Previous issues I commented on look to have been fixed.
- [ ] "Cancelled” should be removed, and "Skipped" should be brought back.

**Review by charlesh88 — APPROVED** [review]
We still need to fix the Activity tab issue, but that is being tracked with #7442.

File: src/plugins/timelist/inspector/svg-progress.js
Comment (akhenry) [inline src/plugins/timelist/inspector/svg-progress.js:1]: This blog post is paywalled. Is there a license associated with this code?

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/plugins/timelist/inspector/svg-progress.js
Comment (akhenry) [inline src/plugins/timelist/inspector/svg-progress.js:1]: Also, this source file needs a copyright statement

**Review by akhenry — CHANGES_REQUESTED** [review]

File: src/plugins/timelist/inspector/svg-progress.js
Comment (shefalijoshi) [reply src/plugins/timelist/inspector/svg-progress.js]: I've removed the progress pie for now. We can add it back. Tracking the commit here: https://github.com/nasa/openmct/commit/fb38cdb60c227aa516c2c83ee9c0de0ab68ad59b

**Review by akhenry — APPROVED** [review]

**Pull request merged**
